Sources/StreamChatUI/ChatMessageList/RETHINK/MessageCell.swift

```diff
@@ -54,17 +58,28 @@ class MessageCell<ExtraData: ExtraDataTypes>: _CollectionViewCell {
     }
 
     private func setUpLayout(options: ChatMessageLayoutOptions) {
+        if options.contains(.flipped) {
+            bubbleView.trailingAnchor
+                .pin(equalTo: contentView.layoutMarginsGuide.trailingAnchor)
+                .isActive = true
+        } else {
+            bubbleView.leadingAnchor
+                .pin(equalTo: contentView.layoutMarginsGuide.leadingAnchor)
+                .isActive = true
+        }
+
+        bubbleView.roundedCorners = options.contains(.continuousBubble) ? .all : CACornerMask.all.excluding(
+            options.contains(.flipped) ? .layerMaxXMaxYCorner : .layerMinXMaxYCorner
+        )
 
         if options.contains(.text) {
-            textView.setContentCompressionResistancePriority(.required, for: .horizontal)
-            textView.setContentCompressionResistancePriority(.required, for: .vertical)
-            containerView.addArrangedSubview(textView)
+            containerView.addArrangedSubview(textView, respectsLayoutMargins: true)
         }
     }
     
     override func updateContent() {
-        // TODO: this shouldn't intialize the text view
-        textView.text = content?.text
+        updateTextView()
+        updateBubbleView()
```

Comment:
These "updateXXX" methods should work the way that they won't initialize the subview if not needed. Only the layout options should have the power of initializing the subviews.

Reply:
hm, but those are private and are called from `updateContent` which is invoked when the view is added to the superview and has access to `uiconfig` 🤔 So it looks there's no chance to call them earlier than expected

Reply:
What I mean is that this way `textView` will be initialized even when layout options don't have `.text`. We want to avoid this to keep everything as fast as possible.

Reply:
@VojtaStavik hey, could you have a look at the latest commit and confirm it's what we want?